name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/OnvistaPDFExtractor.java

```diff
@@ -0,0 +1,533 @@
+                        .assign((t, v) -> {
+                            String notation = v.get("notation");
+                            if (notation != null && !notation.equalsIgnoreCase("STK"))
+                            {
+                                // Prozent-Notierung, Workaround..
+                                t.setShares((asShares(v.get("shares")) / 100));
+                            }
+                            else
+                            {
+                                t.setShares(asShares(v.get("shares")));
+                            }
+                            t.setDate(asDate(v.get("date")));
+                            transferValueMap.put("date", t.getDate());
+                            t.setAmount(asAmount(v.get("amount")));
+                            t.setCurrencyCode(asCurrencyCode(v.get("currency")));
+                        })
+                        
+                        .wrap(t -> new TransactionItem(t));
+        
+        addTaxesSectionsAccount(pdfTransaction);
+        
+        //optional: Reinvestierung in:
+        block = new Block("Die Dividende wurde wie folgt in neue Aktien reinvestiert(.*)");
+        type.addBlock(block);
+        block.set(new Transaction<BuySellEntry>()
+    
+                        .subject(() -> {
+                            BuySellEntry entry = new BuySellEntry();
+                            entry.setType(PortfolioTransaction.Type.BUY);
+                            return entry;
+                        })
+    
+                        .section("name", "isin")
+                        .find("Gattungsbezeichnung ISIN")
+                        .match("(?<name>.*) (?<isin>[^ ]\\S*)$")
+                        .assign((t, v) -> {
+                            t.setSecurity(getOrCreateSecurity(v));
+                        })
+                        
+                        .section("notation", "shares", "amount", "currency")
+                        .find("Nominal Reinvestierungspreis")
+                        //STK 25,000 EUR 0,700000
+                        .match("(?<notation>^EUR|^STK) (?<shares>\\d{1,3}(\\.\\d{3})*(,\\d{3})?) (?<currency>\\w{3}+) (?<amount>\\d{1,3}(\\.\\d{3})*(,\\d{2})?)(.*)")
+                        .assign((t, v) -> {
+                            String notation = v.get("notation");
+                            if (notation != null && !notation.equalsIgnoreCase("STK"))
+                            {
+                                // Prozent-Notierung, Workaround..
+                                t.setShares((asShares(v.get("shares")) / 100));
+                            }
+                            else
+                            {
+                                t.setShares(asShares(v.get("shares")));
+                            }
+                            
+                            t.setDate((LocalDate) transferValueMap.get("date"));
```

Comment:
Ebenfalls eine sehr gute Idee, das hab ich so übernommen und commited.
Für diesen "Spezialfall" hab ich leider auch nur ein PDF-Dokument (das welches auch in Textform für die Tests eingecheckt ist) Aber ich würde sagen, das sollte so funktionieren.
Die transferValueMap hab ich nun entsorgt.